src/pfe/file-watcher/server/src/utils/dockerutil.ts

```diff
@@ -314,20 +314,23 @@ export async function getFilesOrFoldersInContainerWithTimestamp(projectID: strin
         return [];
       } else {
         logger.logInfo("At least one file was found");
-        const files = data.replace(/\n/g, " ").replace(/[^ -~]+/g, "").split(" ").filter((entry: string) => { return entry.trim() != ""; });
+        const lines = data.split("\n");
         const fileIndex = 8; // file is the 9th argument from the above command
         const filesTimestamp: Array<logHelper.LogFiles> = [];
-        for (let index = fileIndex; index < files.length; index = index + fileIndex + 1) {
-          const file = files[index];
-          const dateString = files[index - 3] + " " + files[index - 2] + " " + files[index - 1];
-          if (!folderName) {
-            filesTimestamp.push({file: path.join(fileLocation, file), time: +new Date(dateString)});
-          } else {
-            if (file.toLowerCase() === folderName.toLowerCase()) {
-              filesTimestamp.push({file: path.join(fileLocation, folderName), time: +new Date(dateString)});
-              break;
+        for (let line of lines) {
```

Comment:
Can you explain what changes are being made here?

Reply:
This is a change to read the `ls` output line by line instead of reading them all together and indexing them. Reading them all together and then indexing them may causes issues when the last column on the ls output have spaces...for example, the file might be symlinked to another file and have spaces in the output causing the indexing to be messed up. So a better solution is to use it line by line.